plume-models/src/instance.rs

```diff
@@ -40,80 +40,72 @@ pub struct NewInstance {
 }
 
 impl Instance {
-    pub fn get_local(conn: &Connection) -> Option<Instance> {
+    pub fn get_local(conn: &Connection) -> Result<Instance> {
         instances::table
             .filter(instances::local.eq(true))
             .limit(1)
-            .load::<Instance>(conn)
-            .expect("Instance::get_local: loading error")
+            .load::<Instance>(conn)?
             .into_iter()
-            .nth(0)
+            .nth(0).ok_or(Error::NotFound)
     }
 
-    pub fn get_remotes(conn: &Connection) -> Vec<Instance> {
+    pub fn get_remotes(conn: &Connection) -> Result<Vec<Instance>> {
         instances::table
             .filter(instances::local.eq(false))
             .load::<Instance>(conn)
-            .expect("Instance::get_remotes: loading error")
+            .map_err(Error::from)
     }
 
-    pub fn page(conn: &Connection, (min, max): (i32, i32)) -> Vec<Instance> {
+    pub fn page(conn: &Connection, (min, max): (i32, i32)) -> Result<Vec<Instance>> {
         instances::table
             .order(instances::public_domain.asc())
             .offset(min.into())
             .limit((max - min).into())
             .load::<Instance>(conn)
-            .expect("Instance::page: loading error")
-    }
-
-    pub fn local_id(conn: &Connection) -> i32 {
-        Instance::get_local(conn)
-            .expect("Instance::local_id: local instance not found error")
-            .id
+            .map_err(Error::from)
     }
 
     insert!(instances, NewInstance);
     get!(instances);
     find_by!(instances, find_by_domain, public_domain as &str);
 
-    pub fn toggle_block(&self, conn: &Connection) {
+    pub fn toggle_block(&self, conn: &Connection) -> Result<usize> {
```

Comment:
If I'm not mistaken, the usize is going to be how many entries were affected, which is one?

Reply:
Yes, but https://github.com/Plume-org/Plume/pull/392#discussion_r244369771